modules/survey_accounts/test/survey_accountsTest.php

```diff
@@ -208,14 +208,77 @@ function testSurveyAccountsWithoutPermission()
         $this->resetPermissions();
     }
     /**
-     * Tests that, when add a survey without visit tag
-     * it should appear the error message
+     * Tests that errors are correctly displayed when the user enters incorrect
+     * or invalid values.
+     * The error messages used in this function come from the file
+     * modules/survey_accounts/php/addsurvey.class.inc
      *
      * @return void
      */
     function testSurveyAccountsAddSurvey()
     {
-        //Visit does not exist for given candidate.
+        $visitLabel        = 'V1';
+        $invalidVisitLabel = '';
+        $instrument        = 'bmi';
+
+        // Ensure that an instrument must be supplied.
+        $this->safeGet($this->url . "/survey_accounts/");
+        $btn = self::$add;
+        $this->webDriver->executescript(
+            "document.querySelector('$btn').click()"
+        );
+        $this->safeFindElement(
+            WebDriverBy::Name("CandID")
+        )->sendKeys("999999");
+        $this->safeFindElement(
+            WebDriverBy::Name("PSCID")
+        )->sendKeys("8889");
+        $this->safeFindElement(
+            WebDriverBy::Name("VL")
+        )->sendKeys($visitLabel);
+        $this->safeFindElement(
+            WebDriverBy::Name("fire_away")
+        )->click();
+        $bodyText =  $this->safeFindElement(
+            WebDriverBy::cssSelector(".error")
+        )->getText();
+        $this->assertContains(
+            "Please choose an instrument.",
+            $bodyText
+        );
+
+        // Ensure visit label exists for a candidate.
+        $this->safeGet($this->url . "/survey_accounts/");
+        $btn = self::$add;
+        $this->webDriver->executescript(
+            "document.querySelector('$btn').click()"
+        );
+        $this->safeFindElement(
+            WebDriverBy::Name("CandID")
+        )->sendKeys("999999");
+        $this->safeFindElement(
+            WebDriverBy::Name("PSCID")
+        )->sendKeys("8889");
+        $this->safeFindElement(
+            WebDriverBy::Name("VL")
+        )->sendKeys($visitLabel);
+        $this->safeFindElement(
+            WebDriverBy::Name("Test_name")
+        )->sendKeys($instrument);
+        $this->safeFindElement(
+            WebDriverBy::Name("fire_away")
+        )->click();
+        $bodyText =  $this->safeFindElement(
+            WebDriverBy::cssSelector(".error")
+        )->getText();
+        $this->assertContains(
+            "Visit $visitLabel does not exist for given candidate",
```

Comment:
Travis is failing at this assertion. Here is the error: 
`Failed asserting that 'Visit V2 does not exist for given candidate' contains "Visit V1 does not exist for given candidate".`

I'm unsure why `V2` is being returned since at line 264 you can see that the variable submitted is `$visitLabel`, which is equal to `V1`. 

I can look into this a bit more this afternoon @johnsaigle

Reply:
That would be awesome